### ISSUE TITLE
Idle Connection to `template1` Database

### ISSUE THREAD

**Issue by nnutter** [genome/TestDbServer#56]
We are having [trouble][1] monitoring our TestDbServer deployment because we're connecting to `template1` in [PostgresInstance][2] and apparently that's interfering with Zenoss's ability to monitor PostgreSQL stats.

[1]: https://jira.gsc.wustl.edu/browse/INFOSYS-14357
[2]: https://github.com/genome/TestDbServer/blob/master/lib/TestDbServer/PostgresInstance.pm#L57

**Comment by nnutter** [comment]
I think ultimately we need to change `PostgresInstance` to connect to `test_db` database instead of `template1`.  In addition to the Zenoss issue we would not be able to create databases based on `template1` while connected to it.

```
test_db=> CREATE DATABASE nnutter TEMPLATE template1;
ERROR:  source database "template1" is being accessed by other users
DETAIL:  There is 1 other session using the database.
```

I don't remember the reason we used `template1` in the first place.  The only thing I can think of was concern about the `test_db` existing on a different PostgreSQL server than the ephemeral databases.

This appears to be failing for a different (additional) reason:

```
$ test-db database create --based-on template1
Got error response 400: Bad Request at /usr/bin/test-db-database-create line 15
```

```
[Wed Jan 21 13:03:56 2015] [info] create database from template template1
[Wed Jan 21 13:03:56 2015] [fatal] _create_database_from_template: DBD::Pg::db do failed: ERROR:  must be superuser to alter superusers at lib/TestDbServer/PostgresInstance.pm line 116.
```

I thought we had already [fixed that][1].

[1]: https://github.com/genome/TestDbServer/blob/master/lib/TestDbServer/Command/CreateDatabaseFromTemplate.pm#L34-L36

**Comment by nnutter** [comment]
@brummett, can you chime in when you get a chance?  I'll start working on the change once we decide on a solution.

**Comment by brummett** [comment]
We're connecting to template1 in PostgresInstance because you have to connect to _some_ database.  Yes, test_db could be on a different server, but it's also a security thing.  The credentials to talk to test_db could be different than the admin credentials used to connect to template1 (and then used to create new databases).

The no-brainer thing would be to make yet another template DB (template2?) based on template1 when setting up the server, then change the config item default_template_name to that name.  Is there a better solution?

```_build_admin_dbh()``` in PostgresInstance should probably be changed to use the config value for default_template_name instead of the hard-coded "template1"

**Comment by nnutter** [comment]
I think creating another database for the "admin database" is good.  However, I think the "admin database" and default template need to be separate databases otherwise I don't think you can create a database using the default template while being connected to it.

**Comment by brummett** [comment]
I'm not sure which DB you mean by "admin database".  There's the test_db database with metadata about test databases, template1 used as the default template DB and admin connection, and all the test databases and templates.

You must be able to create a DB using template1 as the the template; that's what PostgresInstance is doing now.  ```_build_admin_dbh()``` is hard-coded to connect to template1, and ```createdb_from_template()``` uses the admin_dbh handle to perform the create database.  Our configuration (I think) and the travis test config have default_template_name set to "template1"

**Comment by brummett** [comment]
We'll definitely need some kind of always-available DB that PostgresInstance can connect to.  My first pass at eliminating the hard-coded "template1" from ```PostgresInstance::_build_admin_dbh()``` was for users to just pass in one of the databases they were already using - the source DB when making a clone.  But that won't work when deleting a database, since you can't delete a DB that has connections.

I'll change it to use whatever the config is for "default_template_name"

**Comment by brummett** [comment]
This is fixed by #58
